src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -366,13 +366,14 @@ private static OptionalLong offsetOfAnonymousRecord(Cursor outermostParent, Curs
     }
 
     public Declaration.Scoped createEnum(Cursor c) {
-        List<Declaration> decls = new ArrayList<>();
-        c.forEach(child -> decls.add(createTree(child)));
         if (c.isDefinition()) {
-            //just a declaration AND definition, we have a layout
-            decls.forEach(d -> {
-                // append declaration string
-                DeclarationString.with(d, enumConstantString(c.spelling(), (Declaration.Constant)d));
+            List<Declaration> decls = new ArrayList<>();
+            c.forEach(child -> {
+                Declaration enumConstantDecl = createTree(child);
+                if (enumConstantDecl != null) {
```

Comment:
Could you add a comment on this `if` that points back at the bug? (just in case we ever wonder why there is a `null` check here in the future)